python/core/qgsgml.sip

```diff
@@ -1,54 +1,93 @@
-#include <qgsgml.h>
+%Docstring
+ This class reads data from a WFS server or alternatively from a GML file. It
+ uses the expat XML parser and an event based model to keep performance high.
+ The parsing starts when the first data arrives, it does not wait until the
+ request is finished *
 %End
 
+%TypeHeaderCode
+#include "qgsgml.h"
+%End
   public:
     QgsGml(
-      const QString& typeName,
-      const QString& geometryAttribute,
-      const QgsFields  &fields );
-
-    ~QgsGml();
-
-    /** Does the Http GET request to the wfs server
-     *  Supports only UTF-8, UTF-16, ISO-8859-1, ISO-8859-1 XML encodings.
-     *  @param uri GML URL
-     *  @param wkbType wkbType to retrieve
-     *  @param extent retrieved extents
-     *  @param userName username for authentication
-     *  @param password password for authentication
-     *  @param authcfg authentication configuration id
-     *  @return 0 in case of success
-     *  @note available in python as getFeaturesUri
-     */
-    int getFeatures( const QString& uri,
-                     QgsWkbTypes::Type* wkbType,
-                     QgsRectangle* extent = 0,
-                     const QString& userName = QString(),
-                     const QString& password = QString(),
+      const QString &typeName,
+      const QString &geometryAttribute,
+      const QgsFields &fields );
+
+    int getFeatures( const QString &uri,
+                     QgsWkbTypes::Type *wkbType,
+                     QgsRectangle *extent = 0,
+                     const QString &userName = QString(),
+                     const QString &password = QString(),
                      const QString &authcfg = QString() ) /PyName=getFeaturesUri/;
+%Docstring
+ Does the Http GET request to the wfs server
+  Supports only UTF-8, UTF-16, ISO-8859-1, ISO-8859-1 XML encodings.
+  \param uri GML URL
+  \param wkbType wkbType to retrieve
+  \param extent retrieved extents
+  \param userName username for authentication
+  \param password password for authentication
+  \param authcfg authentication configuration id
+  :return: 0 in case of success
+.. note::
+
+   available in Python as getFeaturesUri
+ :rtype: int
+%End
 
-    /** Read from GML data. Constructor uri param is ignored
-     *  Supports only UTF-8, UTF-16, ISO-8859-1, ISO-8859-1 XML encodings.
-     */
     int getFeatures( const QByteArray &data, QgsWkbTypes::Type *wkbType, QgsRectangle *extent = 0 );
+%Docstring
+ Read from GML data. Constructor uri param is ignored
+  Supports only UTF-8, UTF-16, ISO-8859-1, ISO-8859-1 XML encodings.
+ :rtype: int
+%End
 
-    /** Get parsed features for given type name */
-    QMap<qint64, QgsFeature* > featuresMap() const;
+    QMap<QgsFeatureId, QgsFeature * > featuresMap() const;
+%Docstring
+Get parsed features for given type name
+ :rtype: QMap<QgsFeatureId, QgsFeature * >
+%End
 
-    /** Get feature ids map */
-    QMap<qint64, QString > idsMap() const;
+    QMap<QgsFeatureId, QString > idsMap() const;
```

Comment:
@nyalldawson @m-kuhn do you know why QgsFeatureId were replaced by qint64 on the SIP files? Any reason not to keep QgsFeatureId?